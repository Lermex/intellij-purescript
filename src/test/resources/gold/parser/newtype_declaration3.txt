Purescript File
  PSProgramImpl(Program)
    PSModuleImpl(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSProperNameImpl(ModuleName)
        PSProperNameImpl(Qualified)
          PsiElement(proper name)('A')
        PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSNewTypeDeclarationImpl(NewtypeDeclaration)
        PsiElement(newtype)('newtype')
        PsiElement(whitespace)(' ')
        PSIdentifierImpl(TypeConstructor)
          PSProperNameImpl(ProperName)
            PsiElement(proper name)('User4')
            PsiElement(whitespace)(' ')
        PSTypeArgsImpl(TypeArgs)
          PSIdentifierImpl(GenericIdentifier)
            PsiElement(identifier)('a')
            PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSIdentifierImpl(TypeConstructor)
          PSProperNameImpl(ProperName)
            PsiElement(proper name)('NoOp3')
            PsiElement(whitespace)(' ')
        PsiElement(identifier)('a')