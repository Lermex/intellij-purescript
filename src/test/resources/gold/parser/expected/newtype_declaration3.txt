Purescript File
  PSModuleDeclarationImpl(MODULE_DECLARATION)
    PsiElement(PSTokenType.module)('module')
    PSModuleNameImpl(MODULE_NAME)
      PsiElement(PSTokenType.PROPER_NAME)('A')
    PsiElement(PSTokenType.where)('where')
  PSDeclarationsImpl(DECLARATIONS)
    PSDeclarationImpl(DECLARATION)
      PSNewTypeDeclarationImpl(NEW_TYPE_DECLARATION)
        PsiElement(PSTokenType.newtype)('newtype')
        PSDeclarationRefTypeImpl(DECLARATION_REF_TYPE)
          PsiElement(PSTokenType.PROPER_NAME)('User4')
        PSDeclarationRefIdListImpl(DECLARATION_REF_ID_LIST)
          PSDeclarationRefIdImpl(DECLARATION_REF_ID)
            PsiElement(PSTokenType.ID)('a')
        PsiElement(PSTokenType.=)('=')
        PSDataDeclarationItemImpl(DATA_DECLARATION_ITEM)
          PSDeclarationRefTypeListImpl(DECLARATION_REF_TYPE_LIST)
            PSDeclarationRefTypeImpl(DECLARATION_REF_TYPE)
              PsiElement(PSTokenType.PROPER_NAME)('NoOp3')
          PSDeclarationRefIdListImpl(DECLARATION_REF_ID_LIST)
            PSDeclarationRefIdImpl(DECLARATION_REF_ID)
              PsiElement(PSTokenType.ID)('a')